cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -87,185 +90,213 @@ public void addStateVariablesToCgmes() {
 
         addVoltagesForBoundaryNodes();
 
-        addPowerFlowToCgmes();
+        addPowerFlows();
 
-        addShuntCompensatorSectionsToCgmes();
+        addShuntCompensatorSections();
 
-        addTapStepToCgmes();
+        addTapSteps();
 
-        addStatusToCgmes();
+        addStatus();
     }
 
     private void addVoltagesForTopologicalNodes() {
-        PropertyBags voltages = new PropertyBags();
-        // add voltages for TpNodes existing in the Model
-        for (PropertyBag tn : cgmes.topologicalNodes()) {
-            Bus b = network.getBusBreakerView().getBus(tn.getId(CgmesNames.TOPOLOGICAL_NODE));
-            PropertyBag p = new PropertyBag(SV_VOLTAGE_PROPERTIES);
+        // Try to add voltages for all TP nodes existing in the CGMES model
+        PropertyBags svVoltages = new PropertyBags();
+        if (cgmes.isNodeBreaker()) {
+            addVoltagesForTopologicalNodesNodeBreaker(svVoltages);
+        } else {
+            // For CGMES bus-branch models:
+            // In IIDM there is a bus at bus-breaker view
+            // for each CGMES topological node
+            cgmes.topologicalNodes().stream().forEach(tp -> {
+                String tpId = tp.getId(CgmesNames.TOPOLOGICAL_NODE);
+                Bus b = network.getBusBreakerView().getBus(tpId);
+                if (b != null) {
+                    svVoltages.add(buildSvVoltage(tp.getId(CgmesNames.TOPOLOGICAL_NODE), b.getV(), b.getAngle()));
+                } else {
+                    // If no IIDM bus has been found it may correspond
+                    // to a boundary, that will be added later
+                    // FIXME(Luma) We should have uniform processing of TP nodes
+                    // present in the original data that have not received an SvVoltage
+                    // this processing should be common to node-breaker and bus-branch
+                }
+            });
+        }
+        cgmes.add(originalSVcontext, "SvVoltage", svVoltages);
+    }
+
+    private void addVoltagesForTopologicalNodesNodeBreaker(PropertyBags svVoltages) {
+        Set<String> processedTPNodes = new HashSet<>();
+        // Try to find an IIDM bus for every TP node
+        // In the CGMES model we have cached
+        // for every IIDM voltage level and node
+        // the original related TP node
+        cgmes.iidmNodesForTopologicalNodes().forEach((iidmVlNode, tp) -> {
+            if (processedTPNodes.contains(tp)) {
+                return;
+            }
+            VoltageLevel vl = network.getVoltageLevel(iidmVlNode.getLeft());
+            int iidmNode = iidmVlNode.getRight().intValue();
+            Bus b = getBusViewBus(vl, iidmNode);
```

Comment:
I think that the use of bus breaker view is more robust and sustanaible ? I have no clear arguments for that, just a feeling.